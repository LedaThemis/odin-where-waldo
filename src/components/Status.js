import '../styles/Status.css';

const Status = ({ text }) => {
  return (
    <div id="status">
      <h3 id="status--text">{text}</h3>
    </div>
  );
};

export default Status;
