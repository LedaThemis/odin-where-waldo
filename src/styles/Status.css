#status {
  position: fixed;
  top: 1rem;
  left: 50%;
  transform: translate(-50%, 0);

  padding: 1rem 4rem;

  text-align: center;
  border-radius: 1rem;

  background-color: #759aab;
  color: white;

  width: 20rem;
}

#status--text {
  margin: 0;
}
