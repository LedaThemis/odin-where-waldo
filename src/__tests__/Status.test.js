import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Status from '../components/Status';
import rerenderer from 'react-test-renderer';

test('should render correctly', () => {
  const tree = rerenderer.create(<Status text={'Sample Text'} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('should render provided text', () => {
  const text = '4witj4gijskdgIJdf';
  render(<Status text={text} />);

  expect(screen.getByText(text)).toBeVisible();
});
