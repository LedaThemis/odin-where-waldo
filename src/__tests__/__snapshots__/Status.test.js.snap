// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  id="status"
>
  <h3
    id="status--text"
  >
    Sample Text
  </h3>
</div>
`;
